### PR TITLE
Add GitHub Issue Templates

### PR DESCRIPTION
**Pull request by DanielYang59**
# Summary

This pull request addresses suggestions made in issue #3449 by adding two new issue templates.

Major changes:

- Added a new issue template for "Bug Report."
- Included an issue template for "Feature Request," aligning it with the original issue template format.

Please feel free to provide any suggestions or feedback.

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/bug_report.yaml [hunk 1]
```diff
@@ -0,0 +1,70 @@
+name: Report a Bug
+description: Use this template to report a Pymatgen-related bug
+title: "[BUG]: "
```
Comment (janosh) [inline .github/ISSUE_TEMPLATE/bug_report.yaml:3]: Can we remove the preset titles?

File: .github/ISSUE_TEMPLATE/feature_request.yaml [hunk 2]
```diff
@@ -0,0 +1,34 @@
+name: Request a Feature
+description: Use this template to request a new feature
+title: "[Feature Request]: "
```
Comment (janosh) [inline .github/ISSUE_TEMPLATE/feature_request.yaml:3]: here as well.

File: .github/ISSUE_TEMPLATE/feature_request.yaml [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+      value:
+    validations:
+      required: true
+
+  - type: textarea
+    id: relevant
+    attributes:
+      label: Relevant Information
+      description: Do you have any other relevant information that might be helpful for understanding or implementing this feature?
+      placeholder: e.g., Additional context about specific use cases, related discussions, or links to relevant literature...
```
Comment (janosh) [inline .github/ISSUE_TEMPLATE/feature_request.yaml:31]: placeholders and descriptions here are a bit verbose. can you cut the language as much as possible?

**Review by janosh — CHANGES_REQUESTED** [review]

File: .github/ISSUE_TEMPLATE/bug_report.yaml (hunk 1 above)
Comment (DanielYang59) [reply .github/ISSUE_TEMPLATE/bug_report.yaml:3]: Yes sure

File: .github/ISSUE_TEMPLATE/feature_request.yaml (hunk 2 above)
Comment (DanielYang59) [reply .github/ISSUE_TEMPLATE/feature_request.yaml:3]: Yes sure (thought the prefix would make it easier to distinguish and also adhere to the original issue template which has the [Feature Request] prefix)

File: .github/ISSUE_TEMPLATE/feature_request.yaml (hunk 3 above)
Comment (DanielYang59) [reply .github/ISSUE_TEMPLATE/feature_request.yaml:31]: Sure I will try. Thanks for the suggestion.

File: .github/ISSUE_TEMPLATE/feature_request.yaml (hunk 2 above)
Comment (janosh) [reply .github/ISSUE_TEMPLATE/feature_request.yaml]: thanks for trying to be consistent! i actually never liked that about the existing feature request template. titles are for content, not for classifying issues. that's what issue labels are for. :)

**Comment by DanielYang59** [comment]
Changes made accordingly. Thanks for your explainations here.

**Review by janosh — APPROVED** [review]
Thanks @DanielYang59 👍

**Comment by DanielYang59** [comment]
@janosh Thanks for reviewing and the comments. Actually my first time PR, just started to working in the community and would try to keep going.

**Comment by janosh** [comment]
That sounds great! The more, the merry! 😄

**Pull request merged**